src/pybind/mgr/dashboard/frontend/src/app/core/mgr-modules/mgr-modules-list/mgr-modules-list.component.ts

```diff
@@ -0,0 +1,163 @@
+import { Component, ViewChild } from '@angular/core';
+
+import { I18n } from '@ngx-translate/i18n-polyfill';
+import { BlockUI, NgBlockUI } from 'ng-block-ui';
+import { timer as observableTimer } from 'rxjs';
+
+import { MgrModuleService } from '../../../shared/api/mgr-module.service';
+import { TableComponent } from '../../../shared/datatable/table/table.component';
+import { CellTemplate } from '../../../shared/enum/cell-template.enum';
+import { CdTableAction } from '../../../shared/models/cd-table-action';
+import { CdTableColumn } from '../../../shared/models/cd-table-column';
+import { CdTableFetchDataContext } from '../../../shared/models/cd-table-fetch-data-context';
+import { CdTableSelection } from '../../../shared/models/cd-table-selection';
+import { Permission } from '../../../shared/models/permissions';
+import { AuthStorageService } from '../../../shared/services/auth-storage.service';
+import { NotificationService } from '../../../shared/services/notification.service';
+
+@Component({
+  selector: 'cd-mgr-modules-list',
+  templateUrl: './mgr-modules-list.component.html',
+  styleUrls: ['./mgr-modules-list.component.scss']
+})
+export class MgrModulesListComponent {
+  @ViewChild(TableComponent)
+  table: TableComponent;
+  @BlockUI()
+  blockUI: NgBlockUI;
+
+  permission: Permission;
+  tableActions: CdTableAction[];
+  columns: CdTableColumn[] = [];
+  modules: object[] = [];
+  selection: CdTableSelection = new CdTableSelection();
+
+  constructor(
+    private authStorageService: AuthStorageService,
+    private mgrModuleService: MgrModuleService,
+    private notificationService: NotificationService,
+    private i18n: I18n
+  ) {
+    this.permission = this.authStorageService.getPermissions().configOpt;
+    this.columns = [
+      {
+        name: this.i18n('Name'),
+        prop: 'name',
+        flexGrow: 1
+      },
+      {
+        name: this.i18n('Enabled'),
+        prop: 'enabled',
+        flexGrow: 1,
+        cellTransformation: CellTemplate.yesNoText
+      }
+    ];
+    const getModuleUri = () =>
+      this.selection.first() && `${encodeURIComponent(this.selection.first().name)}`;
```

Comment:
This looks like you don't have to use template strings here and could instead just write `this.selection.first() && encodeURIComponent(this.selection.first().name)`?

Reply:
You're right, this works. Copied the code from an existing module (user or rbd list).